src/main/java/slimeknights/tconstruct/library/tinkering/ITinkerable.java

```diff
@@ -18,4 +22,32 @@
 
   /** What the tool is made out of. Displayed whet Ctrl is held */
   void getTooltipComponents(ItemStack stack, List<String> tooltips);
+  
+  //From TinkersItem
+  public List<PartMaterialType> getRequiredComponents();
+  //not sure why these both exist but I'll include both just in case
+  public List<PartMaterialType> getToolBuildComponents();
+  
+  public ItemStack buildItemFromStacks(ItemStack[] stacks);
+  
+  public ItemStack buildItem(List<Material> materials);
+  
+  public NBTTagCompound buildItemNBT(List<Material> materials);
```

Comment:
Implementation detail, does not belong into the interface. Should actually be protected in the class.